src/pages/workspace/WorkspacePeoplePage.js

```diff
@@ -0,0 +1,328 @@
+import withWindowDimensions, {windowDimensionsPropTypes} from '../../components/withWindowDimensions';
+
+const propTypes = {
+    ...withLocalizePropTypes,
+
+    ...windowDimensionsPropTypes,
+
+    /** The personal details of the person who is logged in */
+    personalDetails: personalDetailsPropType.isRequired,
+
+    /** The policy passed via the route */
+    policy: PropTypes.shape({
+        /** The policy name */
+        name: PropTypes.string,
+    }),
+
+    /** URL Route params */
+    route: PropTypes.shape({
+        /** Params from the URL path */
+        params: PropTypes.shape({
+            /** policyID passed via route: /workspace/:policyID/people */
+            policyID: PropTypes.string,
+        }),
+    }).isRequired,
+};
+
+const defaultProps = {
+    policy: {
+        name: '',
+    },
+};
+
+class WorkspacePeoplePage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedEmployees: [],
+            isRemoveMembersConfirmModalVisible: false,
+        };
+
+        this.inviteUser = this.inviteUser.bind(this);
+        this.removeUsers = this.removeUsers.bind(this);
+        this.toggleUser = this.toggleUser.bind(this);
+        this.renderItem = this.renderItem.bind(this);
+        this.renderHeader = this.renderHeader.bind(this);
+        this.askForConfirmationToRemove = this.askForConfirmationToRemove.bind(this);
+        this.hideConfirmModal = this.hideConfirmModal.bind(this);
+    }
+
+    /**
+    * Open the modal to invite a user
+    */
+    inviteUser() {
+        Navigation.navigate(ROUTES.getWorkspaceInviteRoute(this.props.route.params.policyID));
+    }
+
+    /**
+    * Remove selected users from the workspace
+    */
+    removeUsers() {
+        // Remove the admin from the list
+        // eslint-disable-next-line max-len
+        removeMembers(this.state.selectedEmployees.filter(email => email !== this.props.session.email), this.props.route.params.policyID);
+        this.setState(prevState => ({
+            ...prevState,
+            selectedEmployees: [],
+            isRemoveMembersConfirmModalVisible: false,
+        }));
+    }
+
+    /**
+    * Show the modal to confirm removal of the selected members
+    */
+    askForConfirmationToRemove() {
+        this.setState(prevState => ({...prevState, isRemoveMembersConfirmModalVisible: true}));
+    }
+
+    /**
+    * Hide the confirmation modal
+    */
+    hideConfirmModal() {
+        this.setState(prevState => ({...prevState, isRemoveMembersConfirmModalVisible: false}));
+    }
+
+    /**
+    * Add or remove user from the selectedEmployees list
+    *
+    * @param {Sting} login
+    */
+    toggleUser(login) {
+        if (login === 'ALL') {
+            this.setState(prevState => ({
+                ...prevState,
+                selectedEmployees: prevState.selectedEmployees.length !== this.props.policy.employeeList.length
```

Comment:
I'm unsure why we should compare the lengths of state and props here to determine which way we should toggle the users. It might be better to look at the state of this checkbox and then update the state based on that.

e.g. 

```js
this.setState(prevState => ({
    shouldSelectAllEmployees: !prevState.shouldSelectAllEmployees,
    selectedEmployees: !prevState.shouldSelectAllEmployees ? this.props.policy.employeeList : [],
}));
```

Reply:
So the reason was that the `all` header checkbox state is not dependant on `this.state. shouldSelectAllEmployees` but on `this.state.selectedEmployees.length === this.props.policy.employeeList.length`.

I think I will actually drop the use of `shouldSelectAllEmployees` as it is making it unnecessarily complex as we would need to check on each toggle if all members have been selected and `shouldSelectAllEmployees` should be set to `true`.

Reply:
That seems fine to me, if we fix up the `toggleUser()` method as well.